src/BayesFilters/src/utils.cpp

```diff
@@ -16,3 +16,106 @@ double bfl::utils::log_sum_exp(const Ref<const VectorXd>& arguments)
 
     return max + log((arguments.array() - max).exp().sum());
 }
+
+
+MatrixXd bfl::utils::quaternion_to_rotation_vector(const Ref<const MatrixXd>& quaternion)
+{
+    MatrixXd rotation_vectors(3, quaternion.cols());
+    for (std::size_t i = 0; i < quaternion.size(); i++)
+    {
+        double norm_n = quaternion.col(i).tail<3>().norm();
+        if (norm_n > std::numeric_limits<double>::epsilon())
+        {
+            const double& w = quaternion.col(i)(0);
```

Comment:
As we do not need the value by reference (we don't really want to keep track of its changes) and being just a double, I suggest to use `const double`.